sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -295,18 +309,20 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
       val requestedColumns =
         projects.asInstanceOf[Seq[Attribute]] // Safe due to if above.
           .map(relation.attributeMap)            // Match original case of attributes.
+          .filterNot(handledSet.contains)
 
       val scan = execution.PhysicalRDD.createFromDataSource(
         projects.map(_.toAttribute),
-        scanBuilder(requestedColumns, pushedFilters),
+        scanBuilder(requestedColumns, candidatePredicates, pushedFilters),
```

Comment:
At here, it is not really necessary to pass in `candidatePredicates` because a data source may reject some filters. We can just pass in the equivalent forms of `pushedFilters`, right?